media/js/firefox/new/yandex/scene1.js

```diff
@@ -0,0 +1,231 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+(function() {
+    'use strict';
+
+    var Yandex = {
+        RUSSIA_COUNTRY_CODE: 'ru',
+        COOKIE_ID: 'firefox-yandex',
+        COOKIE_YANDEX_COHORT: 'firefox-yandex-cohort',
+        COOKIE_EXPIRATION_DAYS: 3,
+        YANDEX_SAMPLE_RATE: 0.9
+    };
+
+    var _client = Mozilla.Client;
+    var _geoTimeout;
+    var _requestComplete = false;
+
+    Yandex.getLocation = function() {
+        // should /country-code.json be slow to load,
+        // just show the regular messaging after 3 seconds waiting.
+        _geoTimeout = setTimeout(Yandex.onRequestComplete, 3000);
+
+        var xhr = new window.XMLHttpRequest();
+
+        xhr.onload = function(r) {
+            // make sure status is in the acceptable range
```

Comment:
If it's not in the acceptable range should we fail faster? Right now it will just timeout (which is fine, that scenario is well handled).

Reply:
Good suggestion!